Fix Profile title when the author has a NIP-05 set

martindsq opened the pull request:
#932 #948 

When the author has a NIP-05, like rabble@nos.social, SwiftUI's Text component detect it has a link and, not only displays it as a link, but also opens the mail app when tapped (even though it is just a Text) basically because Text supports markdown and such. So, what I did was updating the tint color of the navigation bar title text to our custom accent color (not plain blue).

Secondly, the ticket asks for opening njump, not email. This is not a bug because we never intended the title bar to be tappable. However, given that that ticket asks for it, I modified `safeIdentifier` computed property of the Author class so that it returns markdown text taking advantage of the fact that it is only used for displaying it in the title bar and renamed the property to `markdownIdentifier` so it is clear what it does and where it could be used (otherwise I think a developer would think it could be used for identifying authors).

![Simulator Screenshot - iPhone 15 - 2024-03-13 at 16 02 06](https://github.com/planetary-social/nos/assets/1703242/907fa8a0-3751-4e48-947d-069456124c51)

mplorentz commented:
@joshuatbrown yeah I agree. Tappable text in the navbar might confuse people as I've never seen an app do that before.

@martindsq you asked me to link [this](https://github.com/planetary-social/nos/issues/948) bug here as well. Do you want to fix it in this PR?

martindsq commented:
I renamed `markdownIdentifier` to `humanFriendlyIdentifier` because it is not markdown anymore, and just disabled hit testing in the navigation title bar to prevent the app from opening nip05 that are valid urls.

Also fixed #948 because it was a one-liner

martindsq commented:
@joshuatbrown this is ready to be re-reviewed!

joshuatbrown commented on Nos/Views/NosNavigationBar.swift:
```diff
@@ -26,18 +28,28 @@ extension View {
-            .frame(maxWidth: .infinity)
-            .background(Color.appBg)
-            .nosNavigationBar(title: .localizable.homeFeed)
+#Preview {
```
Thanks so much for converting to `#Preview`!

joshuatbrown reviewed (approved):
This looks great!

joshuatbrown commented:
Forgot to mention: also works great on iPhone simulator and Mac!
